src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -91,6 +91,8 @@ public void DisposeInBatch(RenderBatchBuilder batchBuilder)
             {
                 RemoveCascadingParameterSubscriptions();
             }
+
+            DisposeBuffers();
```

Comment:
Any reason you can't merge `Dispose` and `DisposeBuffers` and just call `Dispose` from here, letting it take care of `component.Dispose()`? It would be nice if there was only one place where we call dispose on the component itself.